src/lib.rs

```diff
@@ -1,31 +1,29 @@
-extern crate regex;
-extern crate crypto;
-
-use std::i64;
-use regex::Regex;
-use self::crypto::digest::Digest;
-use self::crypto::sha3::Sha3;
+use crypto::{digest::Digest, sha3::Sha3};
 
 pub fn checksum(address: &str) -> String {
-    let re = Regex::new(r"^0x").unwrap();
-    let address = address.to_lowercase();
-    let address = re.replace_all(&address, "").to_string();
+    let address = address.trim_start_matches("0x").to_lowercase();
+
+    let address_hash = {
+        let mut hasher = Sha3::keccak256();
+        hasher.input(address.as_bytes());
```

Comment:
`input_str` does the same thing under the hood, but we already have `address: String` so it might(?!) be more efficient.